### DISCUSSION TITLE
Error in openHAB 4 from Jinja transform in rtl_433_mqtt_hass.py with "value_template": "{{ float(value) * 99 + 1 }}"

### DISCUSSION THREAD

**Discussion by riegelbrau** [merbanan/rtl_433#2776] [Q&A]
I am using rtl_433_mqtt_hass.py with openHAB 4.04 to auto-create Things from rtl_433. Since some time (don't know exactly when) the Jinja transform throws an error from the battery_ok mapping. The entry "value_template": "{{ float(value) * 99 + 1 }}," leads to `Could not resolve function 'float'`
Log excerpt:
`2024-01-02 13:27:33.347 [DEBUG] [.internal.JinjaTransformationService] - about to transform '1' by the function '{{ float(value) * 99 + 1 }}'`
`2024-01-02 13:27:33.527 [WARN ] [t.generic.ChannelStateTransformation] - Executing the JINJA-transformation failed: An error occurred while transformation. Could not resolve function 'float'`
After some tracing I changed the line to `"value_template": "{{ value|float * 99 + 1 }}",` and it works without any error. 

My questions:

1. Do other users see this error in openHab or Home Assistant?
2. Is there a special reason using the first version of float while other mappings use it like in my change?
3. Does this mapping require a special Jinja version?

Thanks a lot.
Christoph

**Comment by zuckschwerdt** [comment]
And there are also things like `(float(value|float) * 3.6) | round(2)` which really don't look proper.

**Reply by riegelbrau** [reply]
After applying the modification seen above, I faced that my Inkbird ITH20R actually sends `battery_ok = 0.6`:
![grafik](https://github.com/merbanan/rtl_433/assets/73437730/22c73187-8c7b-4ca5-a924-0faedb3ded0f)
This results to many decimal places. Therefore I recommend rounding the result to 0 decimal places for the battery_ok mapping.  My modification looks like this:
`"battery_ok": {
        "device_type": "sensor",
        "object_suffix": "B",
        "config": {
            "device_class": "battery",
            "name": "Battery",
            "unit_of_measurement": "%",
            "value_template": "{{ ( value|float * 99 + 1 ) | round(0) }}",
            "state_class": "measurement",
            "entity_category": "diagnostic"
        }`
